VMBackup/main/hostsnapshotter.py

```diff
@@ -162,8 +162,12 @@ def get_snapshot_info(self, responseBody):
             if(responseBody != None):
                 json_reponseBody = json.loads(responseBody)
                 for snapshot_info in json_reponseBody['snapshotInfo']:
-                    blobsnapshotinfo_array.append(HostSnapshotObjects.BlobSnapshotInfo(snapshot_info['isSuccessful'], snapshot_info['snapshotUri'], snapshot_info['errorMessage'], snapshot_info['statusCode']))
-                    self.logger.log("IsSuccessful:{0}, SnapshotUri:{1}, ErrorMessage:{2}, StatusCode:{3}".format(snapshot_info['isSuccessful'], snapshot_info['snapshotUri'], snapshot_info['errorMessage'], snapshot_info['statusCode']))
+                    self.logger.log("IsSuccessful:{0}, SnapshotUri:{1}, ErrorMessage:{2}, StatusCode:{3}, BlobUri: {4}".format(snapshot_info['isSuccessful'], snapshot_info['snapshotUri'], snapshot_info['errorMessage'], snapshot_info['statusCode'], snapshot_info['blobUri']))
+                    if('DDSnapshotIdentifier' in snapshot_info and snapshot_info['DDSnapshotIdentifier']!= None):
+                        blobsnapshotinfo_array.append(HostSnapshotObjects.BlobSnapshotInfo(snapshot_info['isSuccessful'], snapshot_info['snapshotUri'], snapshot_info['errorMessage'], snapshot_info['statusCode'], snapshot_info['blobUri'], snapshot_info['DDSnapshotIdentifier']))
```

Comment:
convert DDSnapshotIdentifier in the class object and then pass that to BlobSnapshotInfo.
Also, do sanitary testing in non-ultra disk VM to check that the current flow is not broken.
Also, test ultra disk by hard-coding a sample response for ultra disk &  initializing host response to a sample ultra disk response

Reply:
Made the code changes. Testing pending.